### PR TITLE
[PD-3881] [PD-3896] Reporting Fixes

### PR DESCRIPTION
**Pull request by koonsies**
## What's this PR do?
This PR covers a few issues in regards to reporting

1) Back/Forward navigation stopped working after we removed history.listen handling (https://github.com/DirectEmployers/MyJobs/pull/3615 ... yeah, I'm the one who merged it 😭  )
2) When on the Export view for a report, if you select "Export" on another report, the navigation does not work
3) The spinner didn't always show up when it needed to during navigation in Export/Preview.
4) The report list did not populate if the user was on Preview/Export and refreshed their browser.
## Where should the reviewer start?
Reporting.

## How should this be manually tested?
Apart from testing the exact cases mentioned above, check navigation in general. Ensure back, forward, reload and any other combination of navigation works properly and nothing is out of place.

### Front-End

- [ ] Works on IE 11 and Edge
- [ ] Works on Mobile
- [ ] Unit tests added or ticket exists to add tests

### CODE REVIEW

File: gulp/src/reporting/components/ExportReport.jsx [hunk 1]
```diff
@@ -27,15 +29,24 @@ class ExportReport extends Component {
...
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.location !== prevProps.location) {
+      this.loadFromProps();
+    }
+  }
+
+  componentWillUnmount() {
+    this.props.dispatch(highlightReportAction());
```
Comment (koonsies) [inline gulp/src/reporting/components/ExportReport.jsx]: Clear the highlighted report on navigation away.

File: gulp/src/reporting/components/ExportReport.jsx [hunk 2]
```diff
@@ -27,15 +29,24 @@ class ExportReport extends Component {
...
+    if (queryObject.intention) {
+      dispatch(doDataSetMenuFill(queryObject.intention));
     }
-    this.loadData().then(() => dispatch(markPageLoadingAction(false)));
+
+    this.loadFromProps();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.location !== prevProps.location) {
```
Comment (koonsies) [inline gulp/src/reporting/components/ExportReport.jsx]: This is done in Preview, Export, and SetUpReport .. Essentially, check if the history object has changed. Without this, history changes that do not unmount/remount the component will not trigger a reloading of data.

**Comment by aljordan82** [comment]
I'll merge this before the sprint cutoff time

**Comment by koonsies** [comment]
Adding DNM until I can try a new bit of code to completely remove history.listen

**Pull request merged**